README.md

```diff
@@ -199,35 +204,36 @@ sponsoring the project.
 
 ### Type signatures
 
-[Hindley-Milner][Guide:HM] type signatures are used to document functions. You
-might encounter some additional syntax that we use to describe JavaScript
-specific stuff, like [methods](#squiggly-arrows) or functions that take
+The various function signatures are provided in a small language referred to as
+Hindley-Milner notation. You can learn about it in the context of JavaScript
+[here][Guide:HM]. On top of the basic Hindley-Milner notation, we use a few
```

Comment:
Link text should be descriptive. I suggest rewording this so the link text is “Hindley–Milner notation” or similar.

Reply:
:zap: